core/src/main/java/com/sequenceiq/cloudbreak/service/customconfigs/CustomConfigurationsService.java

```diff
@@ -93,16 +129,17 @@ public CustomConfigurations create(CustomConfigurations customConfigurations, St
                         + " exists. Provide a different name"); });
         initializeCrnForCustomConfigs(customConfigurations, accountId);
         customConfigurations.setAccount(accountId);
-        customConfigurations.getConfigurations().forEach(config -> config.setCustomConfigs(customConfigurations));
+        Set<CustomConfigurationProperty> configurationProperties = new HashSet<>(customConfigurations.getConfigurations());
+        configurationProperties.forEach(config -> config.setCustomConfigs(customConfigurations));
         try {
             transactionService.required(() -> {
-                CustomConfigurations created = customConfigurationsRepository.save(customConfigurations);
-                ownerAssignmentService.assignResourceOwnerRoleIfEntitled(ThreadBasedUserCrnProvider.getUserCrn(), created.getCrn(),
+                customConfigurationPropertyRepository.saveAll(configurationProperties);
+                ownerAssignmentService.assignResourceOwnerRoleIfEntitled(ThreadBasedUserCrnProvider.getUserCrn(), customConfigurations.getCrn(),
                         ThreadBasedUserCrnProvider.getAccountId());
-                return created;
+                return customConfigurations;
             });
         } catch (TransactionService.TransactionExecutionException e) {
-            throw new TransactionService.TransactionRuntimeExecutionException(e);
+            throw new InternalServerErrorException(e);
```

Comment:
could you replace this one, with something like `CloudbreakServiceException` or similar? Also please add a message to the new exception